react/src/lib/Popover/index.js

```diff
@@ -216,8 +266,39 @@ class Popover extends React.Component {
               },
               onMouseLeave: e => {
                 e.persist();
-                this.setState({ isHovering: false }, () => this.delayedHide(e));
+                this.handleHide(e);
               },
+              onKeyDown: (e) => {
+                if (this.state.isOpen && this.overlay && this.anchorRef) {
+                  const eventOverlay = ReactDOM.findDOMNode(this.overlay);
+                  const trigger = ReactDOM.findDOMNode(this.anchorRef);
```

Comment:
Do refs have an `isFocused()` method that would prevent us having to look them up in the DOM?